### DISCUSSION TITLE
Missing copyright information in sources of Quarkus

### DISCUSSION THREAD

**Discussion by lechique** [quarkusio/quarkus#25592] [Q&A]
Dear All, 

For usage Quarkus in development within my organization we need to undergo the Open Source Management Office checking. One of items they're checking are copyright information. Unfortunately I can't find any of these in sources and I am afraid this will be not conformant to Apache License in the eyes of our lawyers. Apache license states:

```
To apply the Apache License to your work, attach the following
      boilerplate notice, with the fields enclosed by brackets "[]"
      replaced with your own identifying information. (Don't include
      the brackets!)  The text should be enclosed in the appropriate
      comment syntax for the file format. We also recommend that a
      file or class name and description of purpose be included on the
      same "printed page" as the copyright notice for easier
      identification within third-party archives.

   Copyright [yyyy] [name of copyright owner]
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
       https://www.apache.org/licenses/LICENSE-2.0
```
Can you guide me to any reason for missing copyright statements in the source files? 
Please help me with promoting usage of Quarkus within my organization.

Dominik

**Comment by maxandersen** [comment]
IANAL, but I can tell you that Red Hat legal team advised Quarkus (and several other teams in Red Hat) that the copyright notice is not required to be *in* the files, just clearly visible near the source. I.e. a LICENSE file or similar.

**Reply by lechique** [reply]
Neither am I :) The thing is that LICENSE.txt file has some general information on how copyright notice should look like but then I am missing application of that in the sources. 
Anyway, thank you for the answer. I will try that with my legal department.

**Reply by richardfontana** [reply]
@lechique The quoted part of the Apache License 2.0 document is in an appendix containing a recommendation for license notices. It is not in any sense mandatory and is not part of the operative license. Probably most projects using the Apache License do not follow this recommendation (which in other respects can be seen as outdated at this point). 

Regarding copyright notices, most open source projects do not include them in source files, including undoubtedly most projects that use the Apache License. It can be debated whether there is some value in having some kind of copyright notice in source files, but there's no sense in which they are required as a matter of copyright law in any jurisdiction I am aware of nor by virtue of the license text. Feel free to relay this to your legal department.

**Reply by maxandersen** [reply]
Thanks @richardfontana. 

@lechique fyi, Richard is the red hat legal that provided the guidance we follow.

**Reply by lechique** [reply]
@maxandersen, @richardfontana, thank you for support. Your answers helped me in discussions with legal department.

**Reply by maxandersen** [reply]
Happy it helped you @lechique.

I notice you marked my reply as the answer - could you mark Richards instead ? His is much more authoritative on this than mine :)

**Reply by lechique** [reply]
@maxandersen I don't know how to do that. After unmarking your answer, the "mark as answer" button is at your answer only. Other entries, including one from Richard, seems to be comments to your answer.

**Reply by maxandersen** [reply]
ah ok, seems to be limitation of GitHub's ui. Only top level can be marked.
